File: Solutions in Java/SingleNumber-III.java
/**
 * @author : Anmol Agrawal
 */
class Solution {
    public int[] singleNumber(int[] nums) {
        int xor = 0;
        for(int x : nums){
            xor ^= x;
        }
        String s = Integer.toBinaryString(xor);
        int i = Integer.parseInt(s.substring(Math.max(s.lastIndexOf('1'), 0)), 2);
        int ans[] = new int[2];
        ans[0] = 0;
        for(int x : nums){
            if((i & x) > 0){
                ans[0] ^= x;
            }
        }
        ans[1] = ans[0] ^ xor;
        return ans;
    }
}

File: Solutions in Javascript/SingleNumber-III.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    let xor = 0;
    for(let i = 0; i < nums.length; i++){
        xor ^= nums[i];
    }
    let s = xor.toString(2);
    let i = parseInt(s.substring(Math.max(s.lastIndexOf('1'), 0)), 2);
    let ans = [];
    ans[0] = 0;
    for(let j = 0; j < nums.length; j++){
        if((i & nums[j]) > 0){
            ans[0] ^= nums[j];
        }
    }
    ans[1] = ans[0] ^ xor;
    return ans;
};
